cudamapper/src/overlapper_triggered.cu

```diff
@@ -1,54 +1,196 @@
+
+        Overlap new_overlap;
+
+        new_overlap.query_read_id_  = overlap_end_anchor.query_read_id_;
+        new_overlap.target_read_id_ = overlap_end_anchor.target_read_id_;
+        new_overlap.num_residues_   = overlap.num_residues;
+        new_overlap.target_end_position_in_read_ =
+            overlap_end_anchor.target_position_in_read_;
+        new_overlap.target_start_position_in_read_ =
+            overlap_start_anchor.target_position_in_read_;
+        new_overlap.query_end_position_in_read_ =
+            overlap_end_anchor.query_position_in_read_;
+        new_overlap.query_start_position_in_read_ =
+            overlap_start_anchor.query_position_in_read_;
+        new_overlap.overlap_complete = true;
+
+        // If the target start position is greater than the target end position
+        // We can safely assume that the query and target are template and
+        // complement reads. TODO: Incorporate sketchelement direction value when
+        // this is implemented
+        if (new_overlap.target_start_position_in_read_ >
+            new_overlap.target_end_position_in_read_)
+        {
+            new_overlap.relative_strand = RelativeStrand::Reverse;
+            auto tmp                    = new_overlap.target_end_position_in_read_;
+            new_overlap.target_end_position_in_read_ =
+                new_overlap.target_start_position_in_read_;
+            new_overlap.target_start_position_in_read_ = tmp;
+        }
+        else
+        {
+            new_overlap.relative_strand = RelativeStrand::Forward;
+        }
+        return new_overlap;
+    };
+};
+
+void OverlapperTriggered::get_overlaps(std::vector<Overlap>& fused_overlaps,
+                                       thrust::device_vector<Anchor>& d_anchors,
+                                       const Index& index)
 {
+
     CGA_NVTX_RANGE(profiler, "OverlapperTriggered::get_overlaps");
-    const auto& read_names   = index.read_id_to_read_name();
-    const auto& read_lengths = index.read_id_to_read_length();
-    size_t total_anchors     = anchors.size();
-
-    // fetch memory info of the current device
-    size_t total = 0, free = 0;
-    CGA_CU_CHECK_ERR(cudaMemGetInfo(&free, &total));
-
-    // Using 80% of available memory as heuristic since not all available memory can be used
-    // due to fragmentation.
-    size_t max_anchors_per_block = 0.8 * free / sizeof(Anchor);
-    // The thurst sort function makes a local copy of the array, so we need twice
-    // twice the device memory available for the sort to succeed.
-    max_anchors_per_block /= 2;
-
-    // comparison function object
-    auto comp = [] __host__ __device__(Anchor i, Anchor j) -> bool {
+    const auto& read_names     = index.read_id_to_read_name();
+    const auto& read_lengths   = index.read_id_to_read_length();
+    auto tail_length_for_chain = 3;
+    auto n_anchors             = d_anchors.size();
+
+    // comparison operator - lambda used to compare Anchors in sort
```

Comment:
Can you add a comment about what's the meaning of this comparison?
One can get it by looking into the code, but having one sentence makes the code easier to read

Reply:
This lambda comes from the current implementation, maybe @vellamike can shed more light on the meaning from a domain perspective.